docs/content/preview/api/ysql/the-sql-language/statements/ddl_alter_table.md

```diff
@@ -182,6 +182,65 @@ It quietly succeeds. Now `\d children` shows that the foreign key constraint `ch
 
 Add the specified constraint to the table.
 
+#### [*alter_column_type*]
+
+Change the type of an existing column. If data on disk is required to change, a full table rewrite is needed.
+
+If the optional `COLLATE` clause is not specified, the default collation for the new column type will be used.
+
+If the optional `USING` clause is not provided, the default conversion for the new column value will be the same as an assignment cast from the old type to the new type.
+
+A `USING` clause must be included when there is no implicit assignment cast available from the old type to the new type.
+
+##### Alter type without table-rewrite
+
+If the change doesn't require data on disk to change, concurrent DMLs to the table can be safely performed.
+
+Example:
+
+```sql
+CREATE TABLE test (id BIGSERIAL PRIMARY KEY, a VARCHAR(50));
+ALTER TABLE test ALTER COLUMN a TYPE VARCHAR(51);
+```
+
+##### Alter type with table rewrite
+
+If the change requires data on disk to change, a full table rewrite will be done and the following semantics apply:
+- The action creates an entirely new table under the hood, and concurrent DMLs may not be reflected in the new table which can lead to correctness issues.
+- This action is not compatible with partitioned tables, tables with rules, and tables with CDC streams.
```

Comment:
"not compatible": does it fail or corrupt or crash?

Reply:
See https://github.com/yugabyte/yugabyte-db/pull/17470/commits/186e4996f02df2868d25a4e6d79bf93d99e7b2bf

Reply:
See https://github.com/yugabyte/yugabyte-db/pull/17470/commits/0eeb9e9d30e709c8b0c3e00a875a7675e5219598